### ISSUE TITLE
semaphore_tracker: process died unexpectedly

### ISSUE THREAD

**Issue by ming2055** [heartexlabs/label-studio-ml-backend#122]
uwsgi deploy in docker, webhook trigger fail, Model was not successfully trained

## uwsgi.log

[2022-06-09 08:34:55,150] [DEBUG] [label_studio_ml.api::log_request_info::135] Request body: b'{"action": "PROJECT_UPDATED", "project": {"id": 11, "task_number": 2, "finished_task_number": 2, "total_predictions_number": 1, "total_annotations_number": 2, "num_tasks_with_annotations": 2, "useful_annotation_number": 2, "ground_truth_number": 0, "skipped_annotations_number": 0, "title": "active_learning", "description": "", "label_config": "<View>\\n  <Image name=\\"image\\" value=\\"$image\\"/>\\n  <Choices name=\\"choice\\" toName=\\"image\\">\\n  <Choice value=\\"PNEUMONIA\\"/><Choice value=\\"NORMAL\\"/></Choices>\\n</View>", "parsed_label_config": {"choice": {"type": "Choices", "to_name": ["image"], "inputs": [{"type": "Image", "value": "image"}], "labels": ["PNEUMONIA", "NORMAL"], "labels_attrs": {"PNEUMONIA": {"value": "PNEUMONIA"}, "NORMAL": {"value": "NORMAL"}}}}, "expert_instruction": "", "show_instruction": false, "show_skip_button": true, "enable_empty_annotation": true, "reveal_preannotations_interactively": false, "show_annotation_history": false, "show_collab_predictions": true, "evaluate_predictions_automatically": false, "token": "1a7bb84e0a955da5", "result_count": 0, "color": "#FFFFFF", "maximum_annotations": 1, "min_annotations_to_start_training": 0, "control_weights": {"choice": {"overall": 1.0, "type": "Choices", "labels": {"PNEUMONIA": 1.0, "NORMAL": 1.0}}}, "model_version": "INITIAL", "data_types": {"image": "Image"}, "is_draft": false, "is_published": false, "created_at": "2022-06-08T08:06:51.139867Z", "updated_at": "2022-06-09T08:34:51.242887Z", "sampling": "Sequential sampling", "skip_queue": "REQUEUE_FOR_OTHERS", "show_ground_truth_first": false, "show_overlap_first": false, "overlap_cohort_percentage": 100, "task_data_login": null, "task_data_password": null, "organization": 1, "created_by": 1}}'
/usr/local/lib/python3.7/multiprocessing/semaphore_tracker.py:55: UserWarning: semaphore_tracker: process died unexpectedly, relaunching.  Some semaphores might leak.
  warnings.warn('semaphore_tracker: process died unexpectedly, '
[2022-06-09 08:34:55,157] [INFO] [label_studio_ml.model::run_job::218] Subprocess 75 has been started with args=('PROJECT_UPDATED', {'project': {'id': 11, 'task_number': 2, 'finished_task_number': 2, 'total_predictions_number': 1, 'total_annotations_number': 2, 'num_tasks_with_annotations': 2, 'useful_annotation_number': 2, 'ground_truth_number': 0, 'skipped_annotations_number': 0, 'title': 'active_learning', 'description': '', 'label_config': '<View>\n  <Image name="image" value="$image"/>\n  <Choices name="choice" toName="image">\n  <Choice value="PNEUMONIA"/><Choice value="NORMAL"/></Choices>\n</View>', 'parsed_label_config': {'choice': {'type': 'Choices', 'to_name': ['image'], 'inputs': [{'type': 'Image', 'value': 'image'}], 'labels': ['PNEUMONIA', 'NORMAL'], 'labels_attrs': {'PNEUMONIA': {'value': 'PNEUMONIA'}, 'NORMAL': {'value': 'NORMAL'}}}}, 'expert_instruction': '', 'show_instruction': False, 'show_skip_button': True, 'enable_empty_annotation': True, 'reveal_preannotations_interactively': False, 'show_annotation_history': False, 'show_collab_predictions': True, 'evaluate_predictions_automatically': False, 'token': '1a7bb84e0a955da5', 'result_count': 0, 'color': '#FFFFFF', 'maximum_annotations': 1, 'min_annotations_to_start_training': 0, 'control_weights': {'choice': {'overall': 1.0, 'type': 'Choices', 'labels': {'PNEUMONIA': 1.0, 'NORMAL': 1.0}}}, 'model_version': 'INITIAL', 'data_types': {'image': 'Image'}, 'is_draft': False, 'is_published': False, 'created_at': '2022-06-08T08:06:51.139867Z', 'updated_at': '2022-06-09T08:34:51.242887Z', 'sampling': 'Sequential sampling', 'skip_queue': 'REQUEUE_FOR_OTHERS', 'show_ground_truth_first': False, 'show_overlap_first': False, 'overlap_cohort_percentage': 100, 'task_data_login': None, 'task_data_password': None, 'organization': 1, 'created_by': 1}}, '1654763695')
[2022-06-09 08:34:55,159] [DEBUG] [label_studio_ml.api::log_response_info::140] Response status: 201 CREATED
[2022-06-09 08:34:55,159] [DEBUG] [label_studio_ml.api::log_response_info::141] Response headers: Content-Type: application/json
Content-Length: 24

**Comment by makseq** [comment]
Please provide more details. What model do you use?

**Comment by ming2055** [comment]
> Please provide more details. What model do you use?

I use model pytorch_transfer_learning . When I run it locally，there is no problem. However, when using uWSGI to run in docker, the model cannot be successfully trained.

**Comment by KonstantinKorotaev** [comment]
Hi @ming2055 
Could you please try to use [this docker file](https://github.com/heartexlabs/label-studio-ml-backend/blob/master/label_studio_ml/examples/simple_text_classifier/Dockerfile)?

**Comment by makseq** [comment]
Closed as unanswered issue.
